pkg/api/service/warnings.go

```diff
@@ -41,6 +40,15 @@ func GetWarningsForService(service, oldService *api.Service) []string {
 		}
 	}
 
+	if isHeadlessService(service) && (service.Spec.LoadBalancerIP != "" || len(service.Spec.ExternalIPs) > 0) {
+		if service.Spec.LoadBalancerIP != "" {
+			warnings = append(warnings, "The spec.loadBalancerIP is not supported when the service is set as headless")
```

Comment:
Since these are user-facing I will pick at the words:

"The" is not needed.

"is not supported" is not correct - "is ignored"

"when the service is set as headless" -> "for headless services"